kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ReplaceQuestions.vue

```diff
@@ -1,35 +1,347 @@
+      :cancelText="coreString('cancelAction')"
+      :title="closeConfirmationTitle$()"
+      @cancel="handleCancelClose"
+      @submit="handleConfirmClose"
+    >
+      {{ closeConfirmationMessage$() }}
+    </KModal>
   </div>
 
 </template>
 
 
 <script>
 
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import { getCurrentInstance, computed, ref } from 'kolibri.lib.vueCompositionApi';
+  import { get } from '@vueuse/core';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
   import { injectQuizCreation } from '../../../composables/useQuizCreation';
+  import { PageNames } from '../../../constants/index';
+  import AccordionItem from './AccordionItem';
+  import AccordionContainer from './AccordionContainer';
 
   export default {
     name: 'ReplaceQuestions',
-    setup() {
+    components: {
+      AccordionContainer,
+      AccordionItem,
+    },
+    mixins: [commonCoreStrings],
+    setup(_, context) {
+      const router = getCurrentInstance().proxy.$router;
+      const {
+        replaceQuestions$,
+        deleteSectionLabel$,
+        replaceAction$,
+        selectAllLabel$,
+        closeConfirmationMessage$,
+        closeConfirmationTitle$,
+        replaceQuestionsExplaination$,
+        numberOfSelectedReplacements$,
+        numberOfQuestionsReplaced$,
+      } = enhancedQuizManagementStrings;
       const {
         // Computed
+        activeQuestions,
+        activeSection,
         selectedActiveQuestions,
+        activeResourceMap,
+        replacementQuestionPool,
+        clearSelectedQuestions,
+        toggleItemState,
+        isItemExpanded,
+        selectAllReplacementQuestions,
+        replaceSelectedQuestions,
+        toggleQuestionInSelection,
+        updateSection,
       } = injectQuizCreation();
+
+      const showCloseConfirmation = ref(false);
+
+      function handleCancelClose() {
+        showCloseConfirmation.value = false;
+      }
+
+      function handleConfirmClose() {
+        context.emit('closePanel');
+      }
+
+      const replacements = ref([]);
+
+      function handleReplacement() {
+        const count = replacements.value.length;
+        const questionsNotSelectedToBeReplaced = activeQuestions.value.filter(
+          question => !selectedActiveQuestions.value.includes(question.question_id)
+        );
+        updateSection({
+          section_id: activeSection.value.section_id,
+          questions: [...questionsNotSelectedToBeReplaced, ...replacements.value],
+        });
+        router.replace({
+          name: PageNames.EXAM_CREATION_ROOT,
+          query: {
+            snackbar: numberOfQuestionsReplaced$({ count }),
+          },
+        });
+      }
+
+      function toggleInReplacements(question) {
+        const replacementIds = replacements.value.map(q => q.question_id);
+        if (replacementIds.includes(question.question_id)) {
```

Comment:
My point about the combined exercise_id and question_id seems to be applicable here too.